src/Scaffolding/VS.Web.CG.EFCore/EntityFrameworkModelProcessor.cs

```diff
@@ -554,10 +556,62 @@ private DbContext TryCreateContextUsingAppCode(Type dbContextType, Type startupT
             }
             catch (Exception ex)
             {
+                System.Diagnostics.Debugger.Launch();
+                var exceptionType = ex.GetType();
+                // if MySQL exception with error code 1045, discard error message since it contains sensitive dev information
+                if (exceptionType.Name.Equals(MySqlException, StringComparison.OrdinalIgnoreCase) || exceptionType.FullName.Contains(MySqlException, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (ex.Data.Keys.Count > 0 &&
+                        ex.Data["Server Error Code"] != null &&
```

Comment:
ex.Data.TryGetValue("Server Error Code", out object serverErrorCode)......?
Some dictionary implementations will throw if you try to reference a non-existent key, and then you don't need to repeat the constant.

Reply:
TryGetValue is part of Dictionary and casting [Exception.Data](https://docs.microsoft.com/en-us/dotnet/api/system.exception?view=net-6.0) from  IDictionary (not IDictionary<string,string>) is not pretty. Also the untyped IDictionary property in the Exception does guarantee not throwing on non-existent keys as seen [here](https://docs.microsoft.com/en-us/dotnet/api/system.collections.idictionary.item?redirectedfrom=MSDN&view=net-6.0#System_Collections_IDictionary_Item_System_Object_) in all the dotnet versions we will be supporting. We can still do the cast, just not very pretty.

Reply:
gotcha, you mean if someone overrides the Exception.Data property and it becomes a different implementation. I'll do the cast.